src/modules/formBuilder/formBuilderUtil.ts

```diff
@@ -214,3 +214,79 @@ export const insertItemToDefinition = ({
     newParentArray.splice(insertAtIndex, 0, item);
   }
 };
+
+/**
+ * Validates that the attempted deletion action is allowed. It is allowed if the item
+ * the user is trying to delete doesn't have any other items that refer to it.
+ * If dependents exist, then the operation is not allowed and the function returns a user-facing error string.
+ *
+ * @param toDelete the linkId of the item the user is trying to delete
+ * @param definition the current form definition
+ */
+export const validateDeletion = ({
+  toDelete,
+  definition,
+}: {
+  toDelete: string;
+  definition: FormDefinitionJson;
+}): { success: boolean; errors?: Set<string> } => {
+  // Mapping of linkId of the dependent question => an array of user-facing strings
+  // representing where the problematic dependency is located on that item, e.g. autofill or enableWhen
+  const errors = new Set<string>();
+
+  function checkDependence(
+    toDelete: string,
+    toCheck: FormItem,
+    parentPath: string
+  ) {
+    const label = displayLabelForItem(toCheck);
+    const prefix = parentPath ? `${parentPath} > ${label}` : label;
+
+    toCheck.enableWhen?.forEach((enableWhen) => {
+      if (
+        enableWhen.question === toDelete ||
+        enableWhen.compareQuestion === toDelete
+      ) {
+        errors.add(`${prefix}: <b>conditional visibility</b>`);
+      }
+    });
+
+    toCheck.autofillValues?.forEach((autofillValue) => {
```

Comment:
We're duplicating logic with `buildAutofillDependencyMap` helper here -- sorry I didn't flag that in the spec, I didn't think about that! `buildEnabledDependencyMap` exists as well. 

`buildAutofillDependencyMap` also checks for dependencies in the `formula`, which this doesn't. It might be best to use those helper methods, so we don't have to worry about them falling out of sync.

Hopefully that is an easy change, the only issue I see is that it doesn't provide this "parent path prefix". We would probably need some an additional helper to find the label prefix for a given link id. I wonder if that can be written in a way that is useful if we need it elsewhere, for example:
```
parentLinkIds = findParentLinkIds(itemMap, linkId)   // <== api that would be usable for other needs
prefixedLabel = parentLinkIds.map { |x| itemMap[x].text }.join(" > ") + itemMap[linkId].text
```